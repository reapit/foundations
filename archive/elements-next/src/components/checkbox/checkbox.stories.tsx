import React from 'react'
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0'

import { Checkbox, CheckboxProps } from './checkbox'
import { elFlex } from '@/base/flexbox'
import { elMr4 } from '@/base/spacing'

export default {
  title: 'Checkbox',
  component: Checkbox,
  args: {
    name: 'field_name',
    checked: false,
    id: undefined,
    label: undefined,
    className: undefined,
    disabled: false,
    onChange: e => {
      console.log(e.target.checked)
    },
  },
} as Meta

const Template: Story<CheckboxProps> = args => <Checkbox {...args} />

export const UseAsSingle = Template.bind({})
UseAsSingle.args = {
  checked: true,
  label: 'Checkbox',
}

const GroupTemplate: Story<CheckboxProps> = () => (
  <div className={elFlex}>
    <Checkbox className={elMr4} name="field_name" label="Option 1" value="option_1" />
    <Checkbox className={elMr4} name="field_name" label="Option 2" value="option_2" />
    <Checkbox className={elMr4} name="field_name" label="Option 3" value="option_3" />
    <Checkbox className={elMr4} name="field_name" label="Option 4" value="option_4" />
  </div>
)

export const UseAsGroup = GroupTemplate.bind({})
