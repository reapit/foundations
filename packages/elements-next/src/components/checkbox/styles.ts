import { css } from 'linaria'

export const elCheckbox = css`
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  user-select: none;

  /* Hide the browser's default checkbox */
  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
  }

  /* Create a custom checkbox */
  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
  }

  /* On mouse-over, add a grey background color */
  &:hover input ~ span {
    background-color: #ccc;
  }

  /* When the checkbox is checked, add a blue background */
  & input:checked ~ span {
    background-color: #2196f3;
  }

  /* Create the checkmark/indicator (hidden when not checked) */
  span:after {
    content: '';
    position: absolute;
    display: none;
  }

  /* Show the checkmark when checked */
  & input:checked ~ span:after {
    display: block;
  }

  /* Style the checkmark/indicator */
  & span:after {
    left: 9px;
    top: 5px;
    width: 5px;
    height: 10px;
    border: solid white;
    border-width: 0 3px 3px 0;
    transform: rotate(45deg);
  }
`
