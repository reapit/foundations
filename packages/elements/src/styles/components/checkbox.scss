@import '../base/colors.scss';

.field {
  .checkbox {
    position: absolute;
    width: 0;
    height: 0;
    opacity: 0;

    & + label {
      position: relative;
      cursor: pointer;
      padding: 0;
    }

    // Box.
    & + label:before {
      content: '';
      margin-right: 1rem;
      display: inline-block;
      vertical-align: middle;
      width: 1.6rem;
      height: 1.6rem;
      background: white;
      border: 1px solid $grey;
    }

    & + label.beforeBox:before {
      display: none;
    }

    & + label.beforeBox:after {
      content: '';
      margin-right: 1rem;
      display: inline-block;
      vertical-align: middle;
      width: 1.6rem;
      height: 1.6rem;
      background: white;
      border: 1px solid $grey;
    }

    // Box hover
    &:hover + label:before {
      background: $reapit-teal;
      border-color: $reapit-teal;
    }

    // Box focus
    label:before {
      box-shadow: 0 0 0 1px $grey;
    }

    // Box checked
    &:checked + label:before {
      background: $reapit-teal;
      border-color: $reapit-teal;
    }

    &:disabled + label {
      color: #b8b8b8;
      cursor: auto;
    }

    &:disabled + label:before {
      box-shadow: none;
      background: #ddd;
    }

    // Checkmark
    &:checked + label:after {
      content: '';
      position: absolute;
      left: 8px;
      top: 11px;
      background: white;
      width: 2px;
      height: 2px;
      box-shadow: 2px 0 0 white, 4px 0 0 white, 4px -2px 0 white, 4px -4px 0 white, 4px -6px 0 white, 4px -8px 0 white;
      transform: rotate(45deg);
    }
  }

  .checkbox-error[type='radio'] {
    & + label:before {
      background: white;
      border: 1px solid $reapit-red;
    }
  }

  .radio-wrap {
    margin-top: 4px;
    margin-bottom: 4px;
  }

  .control {
    .subtext {
      font-style: italic;
      display: block;
    }
  }

  &.checkbox-inline {
    .radio-wrap {
      display: inline-block;
      margin-right: 1rem;
    }
  }
}

.field-checkbox {
  position: relative;
  line-height: normal;
}
