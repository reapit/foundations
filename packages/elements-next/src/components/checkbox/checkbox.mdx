---
name: Checkbox
route: /checkbox
---

import { Props } from 'docz'
import DocsWrapper from '@/docz/docs-wrapper'
import Playground from '@/components/playground'
import { Checkbox } from './checkbox'

# Checkbox
Checkboxs are used as the titles of each major section of a page in the interface

## Properties
<Props of={Checkbox} />

## Usage
<Playground>
  <DocsWrapper>
    <Checkbox id="test" name="test" label="test" className="test" value={false} />
  </DocsWrapper>
</Playground>
