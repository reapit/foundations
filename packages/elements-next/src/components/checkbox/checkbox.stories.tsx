import React from 'react'
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0'

import { Checkbox, CheckboxProps } from './checkbox'

export default {
  title: 'Example/Checkbox',
  component: Checkbox,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as Meta

const Template: Story<CheckboxProps> = args => <Checkbox {...args} />

export const Primary = Template.bind({})
Primary.args = {
  value: true,
  label: 'Checkbox',
}

export const Secondary = Template.bind({})
Secondary.args = {
  label: 'Checkbox',
}

export const Large = Template.bind({})
Large.args = {
  label: 'Checkbox',
}

export const Small = Template.bind({})
Small.args = {
  label: 'Checkbox',
}
