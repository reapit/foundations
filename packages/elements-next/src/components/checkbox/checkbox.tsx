import React, { ChangeEvent } from 'react'
import { cx } from 'linaria'
import { elCheckbox } from './styles'

export interface CheckboxProps {
  name: string
  value?: string | ReadonlyArray<string> | number
  checked?: boolean
  id?: string
  label?: string
  className?: string
  disabled?: boolean
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void
}

export const Checkbox: React.FC<CheckboxProps> = ({
  id,
  name,
  label,
  className,
  value,
  checked = false,
  disabled = false,
  onChange,
}) => {
  return (
    <div className={cx(elCheckbox, className)}>
      <input
        disabled={disabled}
        type="checkbox"
        name={name}
        id={id}
        checked={checked}
        value={value}
        onChange={onChange}
      />
      {!!label && <label htmlFor={id}>{label}</label>}
    </div>
  )
}

export default Checkbox
