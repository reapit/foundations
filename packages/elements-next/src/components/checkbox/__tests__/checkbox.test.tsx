import React from 'react'
import { shallow } from 'enzyme'
import { Checkbox } from '../checkbox'

describe('Checkbox', () => {
  it('should match snapshot when true', () => {
    const wrapper = shallow(<Checkbox id="mockName" name="mockName" value={true} className="mockClassName" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<Checkbox name="mockName" label="mockLabel" value={false} disabled={true} />)
    expect(wrapper).toMatchSnapshot()
  })
})
