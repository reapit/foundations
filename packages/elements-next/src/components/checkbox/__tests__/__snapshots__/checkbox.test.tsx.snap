// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<label
  className=""
>
  mockLabel
  <input
    checked={false}
    disabled={true}
    name="mockName"
    type="checkbox"
  />
  <span />
</label>
`;

exports[`Checkbox should match snapshot when true 1`] = `
<label
  className=""
  htmlFor="mockName"
>
  <input
    checked={true}
    disabled={false}
    id="mockName"
    name="mockName"
    type="checkbox"
  />
  <span />
</label>
`;
